mirrord/layer/src/lib.rs

```diff
@@ -448,6 +460,21 @@ fn layer_start(mut config: LayerConfig) {
         config.feature.fs.mode = FsModeConfig::LocalWithOverrides;
     }
 
+    // Check if we're in trace only mode (no agent)
+    let trace_only = std::env::var(TRACE_ONLY_ENV)
+        .unwrap_or_default()
+        .parse()
+        .unwrap_or(false);
+
+    // Disable all features that require the agent
+    if trace_only {
+        config.feature.fs.mode = FsModeConfig::Local;
+        config.feature.network.dns = false;
+        config.feature.network.incoming.mode = IncomingMode::Off;
+        config.feature.network.outgoing.tcp = false;
+        config.feature.network.outgoing.udp = false;
+    }
+
```

Comment:
We can consider extracting this into a function (`fn set_trace_only(mut config: LayerConfig) -> bool`), in order to start making the `layer_start` function a bit shorter.

Reply:
I'd like to make it a real "feature" later on, so maybe do it then?